sdk/servicebus/azure-servicebus/azure/servicebus/_common/utils.py

```diff
@@ -250,3 +250,44 @@ def shutdown(self, wait=True):
         :type wait: bool
         """
         self.executor.shutdown(wait=wait)
+
+
+class AutoComplete(object):
+    """
+    Provides a ContextManager to complete a message on successful exit from the scope,
+    abandoning() otherwise, for instance if an exception is thrown.
```

Comment:
here should be either `abandoning` or `abandon()`?
we don't have a `abandoning()` method :)

Reply:
```suggestion
    calling `abandon()` otherwise, for instance if an exception is thrown.
```
Hahahaha very true.  Going to tweak per the suggestion above (abusing github's in-browser fix for small verbiage like this to avoid branch hopping locally.)